### PR TITLE
M4797 Dynamic repository decorators

### PR DESCRIPTION
**Pull request by bartcharbon**
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] (If you have changed REST API interface) view-swagger.ftl updated
- [x] Test plan template updated
- [x] Clean commits
- [x] Added Feature/Fix to release notes

### CODE REVIEW

File: molgenis-data/src/main/java/org/molgenis/data/decorator/DynamicRepositoryDecoratorRegistryImpl.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		for (DynamicDecorator decorator : decorators)
+		{
+			DynamicRepositoryDecoratorFactory factory = factories.get(decorator.getId());
+			if (factory != null)
+			{
+				repository = factory.createDecoratedRepository(repository);
+			}
+			else
+			{
+				//FIXME: throw appropriate exception
```
Comment (dennishendriksen) [inline molgenis-data/src/main/java/org/molgenis/data/decorator/DynamicRepositoryDecoratorRegistryImpl.java]: Could you resolve this FIXME? In my opinion we should ignore this exception silently.

File: molgenis-data/src/main/java/org/molgenis/data/decorator/meta/DecoratorConfigurationMetadata.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		this.dynamicDecoratorMetadata = requireNonNull(dynamicDecoratorMetadata);
+	}
+
+	@Override
+	public void init()
+	{
+		setLabel("Decorator Configuration");
+		setPackage(decoratorPackage);
+		setDescription(
+				"Configuration entity to configurate which dynamic decorators should be applied to an entity types.");
```
Comment (dennishendriksen) [inline molgenis-data/src/main/java/org/molgenis/data/decorator/meta/DecoratorConfigurationMetadata.java]: confugurate --> configure

Comment (dennishendriksen) [inline molgenis-data/src/main/java/org/molgenis/data/decorator/meta/DecoratorConfigurationMetadata.java]: types --> type

File: molgenis-platform-integration-tests/src/test/java/org/molgenis/integrationtest/platform/DynamicDecoratorIT.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+import static org.molgenis.integrationtest.platform.PlatformIT.waitForWorkToBeFinished;
+import static org.molgenis.security.core.runas.RunAsSystemAspect.runAsSystem;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertTrue;
+
+@ContextConfiguration(classes = { PlatformITConfig.class, AddingRepositoryDecoratorFactory.class,
+		PostFixingRepositoryDecoratorFactory.class })
+public class DynamicDecoratorIT extends AbstractTestNGSpringContextTests
+{
+	private final Logger LOG = LoggerFactory.getLogger(OneToManyIT.class);
```
Comment (dennishendriksen) [inline molgenis-platform-integration-tests/src/test/java/org/molgenis/integrationtest/platform/DynamicDecoratorIT.java]: private final --> private static final

File: molgenis-bootstrap/src/main/java/org/molgenis/bootstrap/RegistryBootstrapper.java [hunk 4]
```diff
@@ -31,6 +32,7 @@
 	private final SystemPackageRegistrar systemPackageRegistrar;
 	private final EntityFactoryRegistrar entityFactoryRegistrar;
 	private final SystemRepositoryDecoratorFactoryRegistrar systemRepositoryDecoratorFactoryRegistrar;
+	private DynamicRepositoryDecoratorFactoryRegistrar dynamicRepositoryDecoratorFactoryRegistrar;
```
Comment (dennishendriksen) [inline molgenis-bootstrap/src/main/java/org/molgenis/bootstrap/RegistryBootstrapper.java]: final

File: molgenis-data/src/main/java/org/molgenis/data/decorator/meta/DynamicDecoratorPopulator.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	{
+		dataService.add(DYNAMIC_DECORATOR,
+				registry.getFactoryIds().filter(this::notPersisted).map(this::createDecorator));
+	}
+
+	private void removeNonExistingDecorators()
+	{
+		Set<String> nonExistingDecorators = getNonExistingDecorators();
+		updateDecoratorConfigurations(nonExistingDecorators);
+		dataService.deleteAll(DYNAMIC_DECORATOR, nonExistingDecorators.stream().map(id -> (Object) id));
```
Comment (dennishendriksen) [inline molgenis-data/src/main/java/org/molgenis/data/decorator/meta/DynamicDecoratorPopulator.java]: ```.map(id -> (Object) id))``` seems unnecessary?

File: molgenis-data/src/test/java/org/molgenis/data/decorator/DynamicRepositoryDecoratorFactoryRegistrarTest.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Mockito.*;
+
+public class DynamicRepositoryDecoratorFactoryRegistrarTest
+{
+
```
Comment (dennishendriksen) [inline molgenis-data/src/test/java/org/molgenis/data/decorator/DynamicRepositoryDecoratorFactoryRegistrarTest.java]: Remove newline

File: molgenis-data/src/test/java/org/molgenis/data/decorator/DynamicRepositoryDecoratorFactoryRegistrarTest.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.context.ApplicationContext;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Mockito.*;
+
+public class DynamicRepositoryDecoratorFactoryRegistrarTest
```
Comment (dennishendriksen) [inline molgenis-data/src/test/java/org/molgenis/data/decorator/DynamicRepositoryDecoratorFactoryRegistrarTest.java]: Please extend from AbstractMockitoTest, and use @Mock for repositoryDecoratorRegistry

File: molgenis-data/src/test/java/org/molgenis/data/decorator/DynamicRepositoryDecoratorRegistryImplTest.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+	private DataService dataService;
+	private EntityType entityType;
+	private RepositoryCollection repositoryCollection;
+	private DecoratorConfiguration decoratorConfiguration;
+	private DynamicRepositoryDecoratorFactory dynamicRepositoryDecoratorFactory;
+
+	@BeforeMethod
+	public void setUp()
+	{
+		repository = mock(Repository.class);
```
Comment (dennishendriksen) [inline molgenis-data/src/test/java/org/molgenis/data/decorator/DynamicRepositoryDecoratorRegistryImplTest.java]: Use @Mock, see e.g. MolgenisLocaleResolverTest

File: molgenis-data/src/test/java/org/molgenis/data/decorator/meta/DynamicDecoratorPopulatorTest.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.stream.Stream;
+
+import static org.mockito.ArgumentCaptor.forClass;
+import static org.mockito.Mockito.*;
+import static org.molgenis.data.decorator.meta.DynamicDecoratorMetadata.DYNAMIC_DECORATOR;
+import static org.testng.Assert.assertEquals;
+
+public class DynamicDecoratorPopulatorTest
+{
+
```
Comment (dennishendriksen) [inline molgenis-data/src/test/java/org/molgenis/data/decorator/meta/DynamicDecoratorPopulatorTest.java]: Remove newline

File: molgenis-data/src/test/java/org/molgenis/data/decorator/meta/DynamicDecoratorPopulatorTest.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.mockito.ArgumentCaptor.forClass;
+import static org.mockito.Mockito.*;
+import static org.molgenis.data.decorator.meta.DynamicDecoratorMetadata.DYNAMIC_DECORATOR;
+import static org.testng.Assert.assertEquals;
+
+public class DynamicDecoratorPopulatorTest
```
Comment (dennishendriksen) [inline molgenis-data/src/test/java/org/molgenis/data/decorator/meta/DynamicDecoratorPopulatorTest.java]: extend AbstractMockitoTest, use @Mock (see previous comment)

**Comment by dennishendriksen** [comment]
```
[INFO] Reactor Summary:
[INFO] 
[INFO] Skipping semantic-mapper
[INFO] This project has been banned from the build due to previous failures.
[INFO] ------------------------------------------------------------------------
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Skipping one-click-importer
[INFO] This project has been banned from the build due to previous failures.
[INFO] ------------------------------------------------------------------------
[INFO] MOLGENIS ........................................... SUCCESS [  1.780 s]
[INFO] test ............................................... SUCCESS [  1.792 s]
[INFO] util ............................................... SUCCESS [  9.016 s]
[INFO] i18n ............................................... SUCCESS [  4.684 s]
[INFO] data ............................................... SUCCESS [ 26.437 s]
[INFO] security-core ...................................... SUCCESS [  5.032 s]
[INFO] settings ........................................... SUCCESS [  3.964 s]
[INFO] data-security ...................................... SUCCESS [ 10.715 s]
[INFO] data-file .......................................... SUCCESS [  4.725 s]
[INFO] data-i18n .......................................... SUCCESS [  5.294 s]
[INFO] scripts-core ....................................... SUCCESS [  4.803 s]
[INFO] js ................................................. SUCCESS [ 12.327 s]
[INFO] data-validation .................................... SUCCESS [ 13.312 s]
[INFO] data-import ........................................ SUCCESS [  8.716 s]
[INFO] jobs ............................................... SUCCESS [ 11.465 s]
[INFO] data-excel ......................................... SUCCESS [  9.115 s]
[INFO] data-csv ........................................... SUCCESS [  6.453 s]
[INFO] amazon-bucket ...................................... SUCCESS [  9.173 s]
[INFO] core ............................................... SUCCESS [  4.582 s]
[INFO] data-cache ......................................... SUCCESS [  9.558 s]
[INFO] data-index ......................................... SUCCESS [ 10.696 s]
[INFO] data-platform ...................................... SUCCESS [  0.336 s]
[INFO] data-plugin ........................................ SUCCESS [  0.175 s]
[INFO] web ................................................ SUCCESS [  8.760 s]
[INFO] security ........................................... SUCCESS [ 15.801 s]
[INFO] core-ui ............................................ SUCCESS [ 21.018 s]
[INFO] apps ............................................... SUCCESS [  9.772 s]
[INFO] data-vcf ........................................... SUCCESS [  8.536 s]
[INFO] data-rest .......................................... SUCCESS [ 15.626 s]
[INFO] genomebrowser ...................................... SUCCESS [  5.209 s]
[INFO] beacon ............................................. SUCCESS [  8.388 s]
[INFO] data-migrate ....................................... SUCCESS [  0.233 s]
[INFO] data-annotators .................................... SUCCESS [ 20.362 s]
[INFO] data-postgresql .................................... SUCCESS [ 10.751 s]
[INFO] bootstrap .......................................... SUCCESS [  0.618 s]
[INFO] dataexplorer ....................................... SUCCESS [ 15.755 s]
[INFO] python ............................................. SUCCESS [  0.226 s]
[INFO] r .................................................. SUCCESS [  3.618 s]
[INFO] searchall .......................................... SUCCESS [  4.455 s]
[INFO] data-elasticsearch ................................. SUCCESS [ 21.675 s]
[INFO] ontology-core ...................................... SUCCESS [  7.184 s]
[INFO] semantic-search .................................... SUCCESS [ 10.057 s]
[INFO] semantic-mapper .................................... SKIPPED
[INFO] ontology ........................................... SUCCESS [  9.696 s]
[INFO] questionnaires ..................................... SUCCESS [  0.809 s]
[INFO] scripts ............................................ SUCCESS [  1.223 s]
[INFO] file-ingester ...................................... SUCCESS [  6.854 s]
[INFO] gavin .............................................. SUCCESS [  9.579 s]
[INFO] fair ............................................... SUCCESS [  8.521 s]
[INFO] swagger ............................................ SUCCESS [  4.293 s]
[INFO] navigator .......................................... SUCCESS [  5.257 s]
[INFO] metadata-manager ................................... SUCCESS [ 11.667 s]
[INFO] one-click-importer ................................. SKIPPED
[INFO] app ................................................ SKIPPED
[INFO] api-tests .......................................... SKIPPED
[INFO] data-rest-client ................................... SUCCESS [  3.436 s]
[INFO] platform-integration-tests ......................... FAILURE [01:17 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 08:22 min
[INFO] Finished at: 2018-01-25T15:40:16+01:00
[INFO] Final Memory: 160M/1810M
[INFO] ------------------------------------------------------------------------
```

**Comment by dennishendriksen** [comment]
```
Failed tests: 
  testAuthorAndBookInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testAuthorAndBookInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testAuthorAndBookInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testAuthorAndBookInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testL1EntitySingleEntityDelete(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testL1EntityStreamingEntityDelete(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testL1SingleEntityUpdate(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testL1StreamingEntityUpdate(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testPersonInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testPersonInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testPersonInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testPersonInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateAuthorOrderAscending(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateAuthorOrderDescending(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateAuthorValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateAuthorValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateAuthorValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateAuthorValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateParentOrderAscending(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateParentOrderDescending(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateParentValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateParentValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateParentValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateParentValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  springTestContextBeforeTestMethod(org.molgenis.integrationtest.platform.importservice.OntologyImportServiceIT): Unable to create SecurityContext using @org.springframework.security.test.context.support.WithMockUser(password=password, value=user, authorities=[], roles=[ENTITY_READ_sys_md_Package, ENTITY_COUNT_sys_md_Package, ENTITY_READ_sys_md_EntityType, ENTITY_COUNT_sys_md_EntityType, ENTITY_READ_sys_md_Attribute, ENTITY_COUNT_sys_md_Attribute, ENTITY_COUNT_sys_ont_OntologyTermDynamicAnnotation, ENTITY_READ_sys_ont_OntologyTermDynamicAnnotation, ENTITY_WRITE_sys_ont_OntologyTermDynamicAnnotation, ENTITY_COUNT_sys_ont_OntologyTermSynonym, ENTITY_READ_sys_ont_OntologyTermSynonym, ENTITY_WRITE_sys_ont_OntologyTermSynonym, ENTITY_COUNT_sys_ont_OntologyTermNodePath, ENTITY_READ_sys_ont_OntologyTermNodePath, ENTITY_WRITE_sys_ont_OntologyTermNodePath, ENTITY_COUNT_sys_ont_Ontology, ENTITY_READ_sys_ont_Ontology, ENTITY_WRITE_sys_ont_Ontology, ENTITY_COUNT_sys_ont_OntologyTerm, ENTITY_READ_sys_ont_OntologyTerm, ENTITY_WRITE_sys_ont_OntologyTerm, ROLE_ENTITY_READ_sys_dec_DecoratorConfiguration], username=ontology_user)

Tests run: 1546, Failures: 25, Errors: 0, Skipped: 1222
```

File: docs/developer_documentation/dynamic-decorators.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
+**
+This section describes how you can make use of the dynamic decorators
+**
+
+Molgenis decorators are used to add functionality, like for example security checks, to repositry functionlity.
```
Comment (dennishendriksen) [inline docs/developer_documentation/dynamic-decorators.md]: Molgenis -> MOLGENIS

Comment (dennishendriksen) [inline docs/developer_documentation/dynamic-decorators.md]: repositry --> repository

Comment (dennishendriksen) [inline docs/developer_documentation/dynamic-decorators.md]: functionlity --> functionality

File: docs/developer_documentation/dynamic-decorators.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+Most of these decorators are apply to all repositories, with the dynamic decorators it is possible to configure for which entity types a decorator should be used.
+
+# Creating a dynamic decorator
+To create a dynamic decorator two classes are needed:
+### The decorator
+This class must extend AbstractRepositoryDecorator, and can be used to override the add, update and delete funtions in this class.
```
Comment (dennishendriksen) [inline docs/developer_documentation/dynamic-decorators.md]: functions --> funtions

File: docs/developer_documentation/dynamic-decorators.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+  @Override
+  public String getId()
+  {
+    return ID;
+  }
+}
+```
+
+# Configuring a dynamic decorator for an entityType
+To apply a dynamic decorator to an entity type, a row shoudl be added to the "sys_dec_DecoratorConfiguration" cia the dataexplorer, this entity contains :
```
Comment (dennishendriksen) [inline docs/developer_documentation/dynamic-decorators.md]: shoudl --> should

Comment (dennishendriksen) [inline docs/developer_documentation/dynamic-decorators.md]: cia --> via

Comment (dennishendriksen) [inline docs/developer_documentation/dynamic-decorators.md]: ' :' --> ':'

**Comment by dennishendriksen** [comment]
Adding a row to the ```sys_dec_DecoratorConfiguration``` table with an invalid entity type identifier should throw an error indicating that the entity type identifier is unknown. Some for updating a row to an invalid entity type identifier.

**Comment by dennishendriksen** [comment]
What is the expected behavior when removing a decorator class: should a decorator configuration row be automatically removed once it has no more decorators?

**Comment by dennishendriksen** [comment]
Logging in with a non-superuser fails, the following error is briefly displayed:
image

**Comment by dennishendriksen** [comment]
Feature change request: extend dynamic decorator metadata with label and description attributes.

**Comment by dennishendriksen** [comment]
Issue: adding a dynamic decorator with an existing id and restarting the application results in one of the two decorators selected. Expected behavior is an error on bootstrap indicating that a dynamic decorator with the given id already exists.

**Comment by dennishendriksen** [comment]
Consistency: Dynamic Decorator vs Decorator Configuration --> Rename Dynamic Decorator to Decorator?

**Comment by bartcharbon** [comment]
It is now no longer possible to add multiple decorators with the same ID
Label and description are added to the dynamic decorator
"Dynamic Decorator" is now called "Decorator"

**Comment by dennishendriksen** [comment]
In case of multiple dynamic decorators the last decorator is called first. We should reverse this list (could do it in a separate PR).

**Comment by dennishendriksen** [comment]
```
Failed tests: 
  testAuthorAndBookInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testAuthorAndBookInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testAuthorAndBookInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testAuthorAndBookInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testL1EntitySingleEntityDelete(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testL1EntityStreamingEntityDelete(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testL1SingleEntityUpdate(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testL1StreamingEntityUpdate(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testPersonInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testPersonInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testPersonInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testPersonInsert(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateAuthorOrderAscending(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateAuthorOrderDescending(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateAuthorValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateAuthorValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateAuthorValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateAuthorValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateParentOrderAscending(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateParentOrderDescending(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateParentValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateParentValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateParentValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
  testUpdateParentValue(org.molgenis.integrationtest.platform.OneToManyIT): No read permission on entity type 'Decorator Configuration' with id 'sys_dec_DecoratorConfiguration'
```

**Pull request merged**